src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/LogHubLoggerProvider.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Logging
+{
+    internal class LogHubLoggerProvider : ILoggerProvider
+    {
+        // Internal for testing
+        internal const string OmniSharpFrameworkCategoryPrefix = "OmniSharp.Extensions.LanguageServer.Server";
+        private const string RazorLanguageServerPrefix = "Microsoft.AspNetCore.Razor.LanguageServer";
+        private const string VSLanguageServerClientPrefix = "Microsoft.VisualStudio.LanguageServerClient.Razor";
+
+        private readonly LogHubLogWriter _logWriter;
+        private readonly ILogger _noopLogger;
+
+        public LogHubLoggerProvider(LogHubLogWriter logWriter)
+        {
+            if (logWriter is null)
+            {
+                throw new ArgumentNullException(nameof(logWriter));
+            }
+
+            _logWriter = logWriter;
+            _noopLogger = new NoopLogger();
+        }
+
+        public ILogger CreateLogger(string categoryName)
+        {
+            if (categoryName is null)
+            {
+                throw new ArgumentNullException(nameof(categoryName));
+            }
+
+            if (categoryName.StartsWith(OmniSharpFrameworkCategoryPrefix, StringComparison.Ordinal))
+            {
+                // Loggers created for O# framework pieces should be ignored for feedback. They emit too much noise.
+                return _noopLogger;
+            }
+
+            if (categoryName.StartsWith(RazorLanguageServerPrefix, StringComparison.Ordinal))
+            {
+                // Reduce the size of the Razor language server categories. We assume nearly all logs here are going to be from the server and thus limiting
+                // the amount of noise they emit will be valuable for readability and will ultimately reduce the size of the log on the users box.
+                categoryName = categoryName.Substring(RazorLanguageServerPrefix.Length + 1 /* . */);
```

Comment:
I was being dumb, lets remove these substrings, they re-allocate strings and given that this is always on I don't want to blow up VS.